qemu/firmware/firmware.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package network provides firmware configurators for use with the Go qemu
+// API.
+package firmware
+
+import (
+	"fmt"
+
+	"github.com/hugelgupf/vmtest/qemu"
+)
+
+// WithOVMF sets the location of the OVMF_CODE.fd and OVMF_VARS.fd firmware
+// files.
+func WithOVMF(vars, code string) qemu.Fn {
+	return func(alloc *qemu.IDAllocator, opts *qemu.Options) error {
+		opts.OvmfCode = code
+		opts.OvmfVars = vars
+		return nil
+	}
+}
+
+// WithUEFI sets the QEMU arguments for enabling UEFI with OVMF firmware.
+func WithUEFI() qemu.Fn {
+	return func(alloc *qemu.IDAllocator, opts *qemu.Options) error {
+		opts.AppendQEMU(
+			"-m", "2G",
```

Comment:
I'd prefer if this were a bit more composable, so without the -m and -machine prescribed. It might be worth naming the min requirements OVMF has as a recommendation to the user in the doc comment.

Reply:
That works for me. But would you like a `WithMemory("2G")` function or are you fine with people specifying this with `AppendQEMU` on their own?

Reply:
I think at the moment I'm leaving it to qemu.ArbitraryArgs -- I wouldn't be against it in the main qemu package